lib/exercism/user_finished_tracks.rb

```diff
@@ -13,20 +13,23 @@ def initialize(user)
   def completed_tracks
     @completed_tracks ||= started_tracks.each_with_object([]) do |db_row, arr|
```

Comment:
It is usually a good idea to let methods do one specific thing, so we could leave the `started_tracks` method as it was before and introduce a new method to check if a track is completed. What do you think of something like:

```ruby
def completed?(track_attributes)
  track = tracks.find { |t| t.id == track_attributes['track_id'] }
  problems_solved = (db_row['completed_problems'].split(',') & track.problems.map(&:slug)).size
  problems_solved >= track.problems.count
end
```

This way we could do:

```ruby
def completed_tracks
  @completed_tracks ||= started_tracks.each_with_object([]) do |db_row, arr|
    arr << track if completed?(db_row)
    arr
  end
end
```

Or even:

```ruby
def completed_tracks
  @completed_tracks ||= started_tracks.keep_if { |db_row| completed?(db_row) }
end

```